### PR TITLE
Rework

### PR DESCRIPTION
**Pull request by franciscoj**
#### 🎩 What?

Rework immortal code in order to prepare it for rails 4.1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,30 +1,38 @@
...
+```
 
 ## Usage
 
-    class User < ActiveRecord::Base
-      include Immortal
-    end
+```ruby
+class User < ActiveRecord::Base
+include Immortal
```
Comment (jhbabon) [inline README.md]: A little bit of indentation here?

File: lib/immortal.rb [hunk 2]
```diff
@@ -1,37 +1,38 @@
...
+        constraint.to_sql.split('AND').reject { |clause| clause.include?('deleted') }
       end.flatten.join(' AND ')
 
       new_scope = new_scope.merge(our_scope.except(:where))
       new_scope = new_scope.where(non_immortal_constraints_sql)
 
-      unscoped.merge(new_scope).scoping do
+      unscoped
+        .merge(new_scope)
+        .scoping do
```
Comment (jhbabon) [inline lib/immortal.rb]: Why this change? It seems a little bit harder to read this multiple line chain and then the `do`.

Comment (franciscoj) [inline lib/immortal.rb]: It helped me while debugging, I'll change it back.

File: lib/immortal.rb [hunk 3]
```diff
@@ -93,20 +92,19 @@ def undeleted_clause_sql
...
       unscoped.where(arel_table[:deleted].eq(true)).constraints.first.to_sql
     end
-
   end
 
   module InstanceMethods
     def self.included(base)
-      unless base.table_exists? && base.columns_hash["deleted"] && !base.columns_hash["deleted"].null
-        Kernel.warn "[Immortal] The 'deleted' column in #{base.to_s} is nullable, change the column to not accept NULL values"
+      unless base.table_exists? && base.columns_hash['deleted'] && !base.columns_hash['deleted'].null
```
Comment (jhbabon) [inline lib/immortal.rb]: Since we are here, maybe we should move this `'deleted'` string hardcoded everywhere to a constant and `#freeze` it.

File: lib/immortal/belongs_to_builder.rb [hunk 4]
```diff
@@ -3,44 +3,39 @@
...
+        assoc.send(:extend, SingularAssociation)
+        assoc.only_deleted_reader(*params)
       end
+    end
 
-      module InstanceMethods
-
-      end #InstanceMethods
-
+    module InstanceMethods
```
Comment (jhbabon) [inline lib/immortal/belongs_to_builder.rb]: Why do we have an empty module here?

File: lib/immortal.rb (hunk 3 above)
Comment (franciscoj) [inline lib/immortal.rb]: I'll iterate on this next (first I wanted just to have the specs)

File: lib/immortal/belongs_to_builder.rb (hunk 4 above)
Comment (franciscoj) [inline lib/immortal/belongs_to_builder.rb]: Who knows...
